src/components/UploadSource/UploadSource.js

```diff
@@ -0,0 +1,19 @@
+import React, { useRef } from 'react';
+import PropTypes from 'prop-types';
+
+export default function UploadSource(props) {
+  return (
+    <div>
+      <input
+        accept='.usfm,.sfm'
+        type='file'
+        onChange={(e) => props.onChange(e.target.files)}
+        multiple
+      />
+    </div>
+  );
+}
+
+UploadSource.propTypes = {
+  onChange: PropTypes.array,
```

Comment:
onChange is of type PropTypes.func,
You can see the browser console in development mode it will throw the errors/warning of such types there